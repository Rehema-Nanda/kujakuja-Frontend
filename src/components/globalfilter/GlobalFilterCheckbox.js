import React from "react";

import { Input, Label, FormGroup } from "reactstrap";
import PropTypes from "prop-types";

export default class GlobalFilterCheckbox extends React.Component {
    constructor(props) {
        super(props);

        this.state = {
            checked: props.selectedItems.indexOf(props.item.id) !== -1,
        };

        this.onChange = this.onChange.bind(this);
        this.label = this.label.bind(this);
    }

    onChange = () => {
        const { checked } = this.state;
        const {
            selectItemHandler, selectItemHandlerArgs, item, allowMultipleServicePointSelect,
        } = this.props;

        this.setState({
            checked: !checked,
        });

        if (allowMultipleServicePointSelect) {
            selectItemHandler(
                item.id,
                ...selectItemHandlerArgs,
            );
        }
        else {
            selectItemHandler(
                item.id,
                ...selectItemHandlerArgs,
                false,
                true,
                false,
            );
        }
    };

    label = () => {
        const { item, itemLabelProperty } = this.props;

        return item[itemLabelProperty];
    };

    render() {
        const { allowMultipleServicePointSelect } = this.props;
        const { checked } = this.state;

        return (
            <FormGroup check>
                <Label check className="map-controls-checkbox">
                    {allowMultipleServicePointSelect
                        ? <Input type="checkbox" checked={checked} onChange={this.onChange} />
                        : <Input type="radio" name="radio1" checked={checked} onChange={this.onChange} />}
                    {this.label()}
                </Label>
            </FormGroup>
        );
    }
}

GlobalFilterCheckbox.propTypes = {
    itemLabelProperty: PropTypes.string.isRequired,
    item: PropTypes.objectOf(PropTypes.any).isRequired,
    selectedItems: PropTypes.arrayOf(PropTypes.any),
    selectItemHandler: PropTypes.func.isRequired,
    selectItemHandlerArgs: PropTypes.arrayOf(PropTypes.any).isRequired,
    allowMultipleServicePointSelect: PropTypes.bool,
};

GlobalFilterCheckbox.defaultProps = {
    selectedItems: [],
    allowMultipleServicePointSelect: true,
};
