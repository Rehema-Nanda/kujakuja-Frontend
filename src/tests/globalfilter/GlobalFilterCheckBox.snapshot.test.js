import React from 'react';
import renderer from 'react-test-renderer';
import GlobalFilterCheckbox from '../../components/globalfilter/GlobalFilterCheckbox';

it('renders correctly', () => {
    const tree = renderer.create(
        <GlobalFilterCheckbox
            item={{id: 1, name: 'Uganda'}}
            itemLabelProperty="name"
            selectItemHandler={() => {}}
            selectItemHandlerArgs={[]}
            selectedItems={[]}
        />
    ).toJSON();

    expect(tree).toMatchSnapshot();
});
