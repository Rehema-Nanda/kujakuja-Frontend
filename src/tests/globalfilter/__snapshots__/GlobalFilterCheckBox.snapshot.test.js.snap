// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="position-relative form-check"
>
  <label
    className="map-controls-checkbox form-check-label"
  >
    <input
      checked={false}
      className="form-check-input"
      onChange={[Function]}
      type="checkbox"
    />
    Uganda
  </label>
</div>
`;
